packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,297 @@
+      "getNextTextRange",
+      "getHyperlinkRanges",
+      "getLastOrNullObject",
+      "getDescendants",
+      "getAncestorOrNullObject",
+      "getAncestor",
+      "getByIdOrNullObject",
+      "getById",
+      "getLevelString",
+      "getLevelParagraphs",
+      "getBase64ImageSrc",
+      "getSelection",
+      "getByTypes",
+      "getByTag",
+      "getReferenceId", 
+      "getDocument",
+    ]);
+
+    function getPropertyThatHadToBeLoaded(node: TSESTree.Node): string | undefined {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent?.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+        return node.parent.property.name;
+      }
+      return undefined;
+    }
+
+    function isLoadFunction(node: TSESTree.Node): boolean {
+      return (node.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.property.type === TSESTree.AST_NODE_TYPES.Identifier
+        && node.parent.property.name === "load");
+    }
+
+    function callsGetAPIFunction(node: TSESTree.Identifier): boolean {
+      return getFunctions.has(node.name);
+    }
+
+    function isVariableDeclaration(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.VariableDeclarator) {
+          return true;
+      }
+      return false;
+    }
+
+    function isGetVariableDeclaration(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.VariableDeclarator
+        && node.parent.init?.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.init.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.init.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.parent.init.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function isAssignmentExpression(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.AssignmentExpression) {
+          return true;
+      }
+      return false;
+    }
+
+    function isGetAssignmentExpression(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.AssignmentExpression
+        && node.parent.right.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.right.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.right.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.parent.right.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function getLoadedPropertyName(node: TSESTree.Node): string {
+      if(node.parent?.parent?.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.parent.arguments[0].type === TSESTree.AST_NODE_TYPES.Literal) {
+          return node.parent.parent.arguments[0].value as string;
```

Comment:
You should simplify instead of having to look at parents of parents ect as that narrows the use of this method to a very specific type of heirachy.

Reply:
I didn't quite get it, what do you mean by "ect"? Was it a typo?

Reply:
That should have been "etc" . . . short hand for etcetera.  Basically I disagree with directly traveling up the tree so much in this way.